subsys/net/lib/http/http_server_http1.c

```diff
@@ -256,6 +257,135 @@ static int dynamic_post_req(struct http_resource_detail_dynamic *dynamic_detail,
 	return 0;
 }
 
+#if defined(CONFIG_FILE_SYSTEM)
+
+HTTP_SERVER_CONTENT_TYPE(html, "text/html")
+HTTP_SERVER_CONTENT_TYPE(css, "text/css")
+HTTP_SERVER_CONTENT_TYPE(js, "text/javascript")
+HTTP_SERVER_CONTENT_TYPE(jpg, "image/jpeg")
+HTTP_SERVER_CONTENT_TYPE(png, "image/png")
+HTTP_SERVER_CONTENT_TYPE(svg, "image/svg+xml")
+
+static void _get_content_type_from_extension(char *url, size_t url_len, char *content_type,
```

Comment:
BTW, no need to prefix static functions with `_`, I also recall that the single underscore is meant to be used in C-library only. Anyway, there is probably some tool that will start to complain about these at some point so better to remove the underscore from function name.

Reply:
As it is needed for http2 as well I renamed it to `http_server_get_content_type_from_extension` and moved it to `http_server_core.c`.